src/adapter/src/config/frontend.rs

```diff
@@ -34,15 +35,44 @@ pub struct SystemParameterFrontend {
 impl SystemParameterFrontend {
     /// Construct a new [SystemParameterFrontend] instance.
     pub fn new(
+        env_id: EnvironmentId,
         ld_sdk_key: &str,
-        ld_context_key: &str,
         ld_key_map: BTreeMap<String, String>,
     ) -> Result<Self, anyhow::Error> {
         let ld_config = ld::ConfigBuilder::new(ld_sdk_key).build();
         let ld_client = ld::Client::build(ld_config)?;
-        let ld_ctx = ld::ContextBuilder::new(ld_context_key)
-            .build()
-            .map_err(|e| anyhow::anyhow!(e))?;
+        let ld_ctx = if env_id.cloud_provider() != "local" {
+            // We don't use the Display for EnvironmentId implementation here
+            // because we don't want to have the ordinal as part of the LD key.
+            //
+            // *Warning*: once the LD integration is live, our contexts will be
+            // populated with the key synthesized below. Any changes to that key
+            // in the backend (that is, here) will also have to be reflected in
+            // the feature targeting config in LD, otherwise environments might
+            // receive a different configs upon restart.
+            let ld_ctx_key = format!(
+                "{}-{}-{}",
+                env_id.cloud_provider(),
+                env_id.cloud_provider_region(),
+                env_id.organization_id(),
+            );
+            ld::ContextBuilder::new(ld_ctx_key)
+                // TODO: uncomment one once the EAP for contexts is enabled in LD
+                // .kind("environment")
+                // .set_string("cloud_provider", env_id.cloud_provider())
+                // .set_string("cloud_provider_region", env_id.cloud_provider_region())
+                // .set_string("organization_id", env_id.organization_id().to_string())
```

Comment:
What happens if you set these without the EAP enabled in LD? Does the server refuse the connection?

Reply:
Without setting `.kind("environment")` the `.set_string(...)` lines are dropped because the context is assumed to be of kind  `user`. Once we have access to the EAP features we can create a new context kind `environment` that has all `EnvironmentId` fields set as separate `.set_string(...)` calls. That way it will be a bit less clumsy to target say a specific `organization_id` (at the moment you need to do a pattern match rule on the key).

Reply:
I was mainly wondering if it would be fine to comment-in these lines already now and they just get ignored until we get EAP access. But sounds like that `kind` setting would be problematic.